### PR TITLE
Add :hover styleRule for modebar.bgcolor

### PR DESCRIPTION
**Pull request by etpinard**
... and a `.ease-bg` class to match modebar button opacity transition.

fixes https://github.com/plotly/plotly.js/issues/3388  - for example:

```js
Plotly.newPlot('graph', [{
  y: [1, 2, 1]
}], {
  width: 500,
  paper_bgcolor: '#d3d3d3',
  margin: {t: 0},
  modebar: {bgcolor: 'red'}
}, {
  watermark: true
})
```
peek 2019-01-04 12-58

cc @plotly/plotly_js - I would appreciate if modebar-expert @antoinerg would test this one out. Thanks!

### CODE REVIEW

**Comment by etpinard** [comment]
Ping @antoinerg - could you take a look at this today. I'll :heart:  to include this in 1.43.2. Thanks!

File: src/components/modebar/modebar.js [hunk 1]
```diff
@@ -52,21 +52,22 @@ proto.update = function(graphInfo, buttons) {
...
 
+    var style = fullLayout.modebar;
+    var bgSelector = context.displayModeBar === 'hover' ? '.js-plotly-plot .plotly:hover ' : '';
+
     Lib.deleteRelatedStyleRule(modeBarId);
-    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId, 'background-color: ' + fullLayout.modebar.bgcolor);
-    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn .icon path', 'fill: ' + fullLayout.modebar.color);
-    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn:hover .icon path', 'fill: ' + fullLayout.modebar.activecolor);
-    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn.active .icon path', 'fill: ' + fullLayout.modebar.activecolor);
+    Lib.addRelatedStyleRule(modeBarId, bgSelector + '#' + modeBarId, 'background-color: ' + style.bgcolor);
```
Comment (antoinerg) [inline src/components/modebar/modebar.js]: Would this work without `bgSelector`? `modeBarId` should already be unique so this might not be necessary.

Comment (etpinard) [reply src/components/modebar/modebar.js]: Maybe there's a way (not a CSS expert here), but I think that `:hover` pseudo-class must be tagged onto the graph div class.

File: test/jasmine/tests/modebar_test.js [hunk 2]
```diff
@@ -1336,16 +1336,55 @@ describe('ModeBar', function() {
...
+                style = window.getComputedStyle(gd._fullLayout._modeBar.element);
+                expect(style.backgroundColor).toBe('rgba(0, 0, 0, 0)');
+                expect(getStyleRule().rules[3].style.backgroundColor).toBe(colors[1]);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('changes background color (displayModeBar: true)', function(done) {
+            function getStyleRule() {
```
[the remark this answers is not in the document]
Comment (etpinard) [reply test/jasmine/tests/modebar_test.js]: Good call. Done in -> a4fb8cd37

File: src/components/modebar/modebar.js (hunk 1 above)
Comment (antoinerg) [reply src/components/modebar/modebar.js]: Ok thanks for the explanation. This makes sense!

**Comment by antoinerg** [comment]
Thanks for the fix and the nice tests @etpinard :dancer:

**Pull request merged**